src/extension/background-script/connectors/lawallet.ts

```diff
@@ -0,0 +1,540 @@
+import { schnorr } from "@noble/curves/secp256k1";
+import * as secp256k1 from "@noble/secp256k1";
+import type { ResponseType } from "axios";
+import { Method } from "axios";
+import lightningPayReq from "bolt11";
+import Hex from "crypto-js/enc-hex";
+import sha256 from "crypto-js/sha256";
+import { nip04, relayInit, type Relay } from "nostr-tools";
+import { Event, EventKind } from "~/extension/providers/nostr/types";
+import { Account } from "~/types";
+
+import toast from "~/app/components/Toast";
+import { getEventHash } from "~/extension/background-script/actions/nostr/helpers";
+import Nostr from "~/extension/background-script/nostr";
+import Connector, {
+  CheckPaymentArgs,
+  CheckPaymentResponse,
+  ConnectPeerResponse,
+  ConnectorTransaction,
+  GetBalanceResponse,
+  GetInfoResponse,
+  GetTransactionsResponse,
+  KeysendArgs,
+  MakeInvoiceArgs,
+  MakeInvoiceResponse,
+  SendPaymentArgs,
+  SendPaymentResponse,
+  SignMessageArgs,
+  SignMessageResponse,
+} from "./connector.interface";
+
+interface Config {
+  privateKey: string;
+  apiEndpoint: string;
+  identityEndpoint: string;
+  ledgerPublicKey: string;
+  urlxPublicKey: string;
+  relayList: string[];
+  relayUrl: string;
+}
+
+export class HttpError extends Error {
+  status: number;
+  error: Error | undefined;
+
+  constructor(status: number, message: string, error?: Error) {
+    super(message);
+    this.status = status;
+    this.error = error;
+  }
+}
+
+export default class LaWallet implements Connector {
+  account: Account;
+  config: Config;
+  relay: Relay;
+  public_key: string;
+  access_token?: string;
+  access_token_created?: number;
+  refresh_token?: string;
+  refresh_token_created?: number;
+  noRetry?: boolean;
+
+  invoices_paid: InvoiceCache = {};
+  last_invoice_check: number = 0;
+
+  constructor(account: Account, config: Config) {
+    this.account = account;
+    this.config = config;
+    this.public_key = new Nostr(config.privateKey).getPublicKey();
+    this.relay = relayInit(config.relayUrl);
+  }
+
+  async init() {
+    return Promise.resolve();
+  }
+
+  unload() {
+    this.relay.close();
+    return Promise.resolve();
+  }
+
+  get supportedMethods() {
+    return [
+      "getInfo",
+      "makeInvoice",
+      "sendPayment",
+      "signMessage",
+      "getInvoices",
```

Comment:
remote getInvoices from here also from the code. add getTransactions in the list of supported methods

Reply:
Do you want me to remove getInvoices aswell. I removed the private method `getPayments`.

Reply:
as its not used anywhere. and it's not a standard connector interface method. getTransactions is the declared method in connector interface. are we using getInvoices anywhere?

Reply:
done